src/dataframe/dataframe.jl

```diff
@@ -1223,7 +1321,18 @@ function Base.push!(df::DataFrame, row::Union{AbstractDict, NamedTuple}; cols::S
             else
                 val = row[nm]
             end
-            push!(col, val)
+            S = typeof(val)
+            T = eltype(col)
+            V = promote_type(S, T)
+            if S <: T || V <: T || !promote
+                # if S <: T || V <: T this should never throw an exception
```

Comment:
This comment applies in other similar places, right?

Reply:
yes - should I add it?

Reply:
I actually had it in most places. Added in two more.